android/src/androidTest/java/io/ably/lib/test/android/AndroidPushTest.java

```diff
@@ -1214,24 +1242,31 @@ public void run() throws Exception {
 						registerCallback.waitFor();
 						assertNull(registerCallback.error);
 					} else {
-						// RSH3d3b
 						requestWaiter.waitFor();
 						Helpers.RawHttpRequest request = requestWaiter.result;
-						assertEquals("PATCH", request.method);
 						assertEquals("/push/deviceRegistrations/"+testActivation.rest.push.getLocalDevice().id, request.url.getPath());
-						assertEquals(
-								JsonUtils.object()
-										.add("push", JsonUtils.object()
-												.add("recipient", JsonUtils.object()
-														.add("transportType", "gcm")
-														.add("registrationToken", updatedRegistrationToken))).toJson().toString(),
-								Serialisation.msgpackToGson(request.requestBody.getEncoded()).toString());
-						String authToken = Helpers.tokenFromAuthHeader(request.authHeader);
+						String authToken = Base64Coder.decodeString(request.requestHeaders.get("X-Ably-DeviceToken").get(0));
 						assertEquals(testActivation.rest.push.getLocalDevice().deviceIdentityToken, authToken);
+
+						JsonObject requestBody = (JsonObject)Serialisation.msgpackToGson(request.requestBody.getEncoded());
+						JsonObject requestRecipient = requestBody.getAsJsonObject("push").getAsJsonObject("recipient");
+						assertEquals("fcm", requestRecipient.getAsJsonPrimitive("transportType").getAsString());
+						assertEquals(updatedRegistrationToken, requestRecipient.getAsJsonPrimitive("registrationToken").getAsString());
+
+						if(isExpectingRegistrationValidation) {
+							// RSH3f1a: PUT the entire DeviceDetails
+							assertEquals("PUT", request.method);
+							assertTrue(requestBody.has("deviceSecret"));
+							assertTrue(requestBody.has("clientId"));
+						} else {
+							// RSH3d3b: PATCH the updated members
```

Comment:
So, the deleted assertion for method == PATCH should be moved here instead?
```suggestion
							// RSH3d3b: PATCH the updated members
							assertEquals("PATCH", request.method);
```